src/general/CombinableActionData.cs

```diff
@@ -13,6 +13,11 @@ public abstract class CombinableActionData
     /// </summary>
     public virtual bool ResetsHistory => false;
 
+    /// <summary>
+    ///   Should this action be merged with the previous one if possible?
+    /// </summary>
+    public virtual bool WantsMerge => false;
```

Comment:
I think this should be a method like:
```c#
public virtual bool WantsMergeWith(CombinadleActionData other){
return false;
}
```
then child classes that support merging can return true if the other object is of the right type. This will give much more flexibility to the design which I foresee the chance of needing.